test/format/snap.c

```diff
@@ -659,7 +659,8 @@ snap_repeat(TINFO *tinfo, SNAP_OPS *snap)
 
         testutil_check(session->rollback_transaction(session, NULL));
     }
-    testutil_assert(max_retry < MAX_RETRY_ON_ROLLBACK);
+    if (max_retry >= MAX_RETRY_ON_ROLLBACK)
+        WARN("%s", "snap repeat exceeds maximum retry.");
```

Comment:
Could you make this more precise by keying it off the string returned from `WT_SESSION::get_rollback_reason`.

I think it's reasonable for it to fail if the rollback is a repeated rollback due to a transaction conflict.

It also deserves a comment - explaining why it can happen sometimes.

Reply:
done